src/nlp/arrow_writer.py

```diff
@@ -183,17 +185,23 @@ def _build_writer(self, inferred_schema: pa.Schema):
         if self.disable_nullable:
             self._schema = pa.schema(pa.field(field.name, field.type, nullable=False) for field in self._schema)
         if self.with_metadata:
-            self._schema = self._schema.with_metadata(self._build_metadata(DatasetInfo(features=self._features)))
+            self._schema = self._schema.with_metadata(
+                self._build_metadata(DatasetInfo(features=self._features), self.fingerprint)
```

Comment:
In my early implementation I added the `fingerprint` field directly in the `DatasetInfo` structure.

What do you think?

Reply:
The fingerprint changes across machines as it sometimes hashes absolute paths.
I'm not sure this should be inside `DatasetInfo` that are meant to be shared